pkg/block/fetcher.go

```diff
@@ -561,6 +561,8 @@ func NewDeduplicateFilter() *DeduplicateFilter {
 // Filter filters out duplicate blocks that can be formed
 // from two or more overlapping blocks that fully submatches the source blocks of the older blocks.
 func (f *DeduplicateFilter) Filter(_ context.Context, metas map[ulid.ULID]*metadata.Meta, synced *extprom.TxGaugeVec, _ bool) error {
+	f.duplicateIDs = nil
```

Comment:
Can we reuse array? (:

```suggestion
	f.duplicateIDs = f.duplicateIDs[:0]
```

Reply:
As per: https://thanos.io/contributing/coding-style-guide.md/#reuse-arrays